src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -790,6 +799,9 @@ private void StartSendBlob(byte[] incoming, int count, AsyncProtocolRequest asyn
                 }
                 else
                 {
+                    // Throw before starting async write
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Why this rather than passing the cancellation token into the WriteAsync call?

Reply:
fixed.